mautic/mautic
# Webhook email content exclusion

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [✅]
| New feature/enhancement? (use the a.x branch)      | [✅]
| Deprecations?                          | [❌]
| BC breaks? (use the c.x branch)        | [❌]
| Automated tests included?              | [✅] 
| Related user documentation PR URL      |  
| Related developer documentation PR URL |  
| Issue(s) addressed                     |  



[//]: # ( Required: )
#### Description:
This PR introduces a new configuration toggle `Send email details` in the webHook settings that is enabled by default. If this toggle is disabled, only a limited set of data is sent for `Email Open Event` and `Email Send Event` webHook events.

This PR also fixes JMS API meta driver configuration (via a32d982c9c5f675042aaad2b278999c8ef22a13e) that got broken in https://github.com/mautic/mautic/pull/12396.

#### Steps to test this PR:
1. Create a new webHook with `Email Open Event` and `Email Send Event` checkboxes checked. (You can use service like https://webhook.site/ to read webHook bodies)
2. Invoke these events by sending and opening an email.
3. Check the body of received webHooks(Can be split into multiple webHooks). They should contain the keys as follows:
  - `mautic.email_on_send.email.customHtml`
  - `mautic.email_on_send.email.plainText`
  - `mautic.email_on_send.content`
  - `mautic.email_on_send.tokens`
  - `mautic.email_on_open.stat.email.customHtml`
  - `mautic.email_on_open.stat.email.plainText`
4. Turn off `Send email details` in the webHook settings.
![102767203-7bb06f00-437f-11eb-8932-b590923cae09](https://github.com/mautic/mautic/assets/6388925/b1a821b8-0f47-4d9a-9712-6cbce55b4323)
5. Invoke these events again by sending and opening another email.
6. Check the body of received webHooks(Can be split into multiple webHooks). They should no longer contain the keys mentioned in the 3. step.
